editor.planx.uk/src/@planx/components/FileUploadAndLabel/SelectMultiple.tsx

```diff
@@ -1,233 +1,318 @@
+ */
+const renderGroup: AutocompleteProps<
+  Option,
+  true,
+  true,
+  false,
+  "div"
+>["renderGroup"] = ({ group, key, children }) => (
+  <List key={`group-${key}`} role="group" sx={{ paddingY: 0 }}>
+    <ListSubheader
+      role="presentation"
+      sx={(theme) => ({
+        borderTop: 1,
+        borderBottom: 1,
+        borderColor: theme.palette.border.main,
+      })}
+    >
+      <Typography py={1} variant="subtitle2" component="h4">
+        {`${capitalize(group)} files`}
+      </Typography>
+    </ListSubheader>
+    {children}
+  </List>
+);
+
+/**
+ * Function which returns the options (li elements) used by Autocomplete
+ */
+const renderOption: AutocompleteProps<
+  Option,
+  true,
+  true,
+  false,
+  "div"
+>["renderOption"] = (props, option, { selected }) => (
+  <ListItem {...props}>
+    <Checkbox
+      data-testid="select-checkbox"
+      inputProps={{
+        "aria-label": `${option.name}`,
+      }}
+      checked={selected}
+    />
+    <ListItemText>{option.name}</ListItemText>
+  </ListItem>
+);
+
+const PopupIcon = <ArrowIcon sx={{ color: "primary.main" }} fontSize="large" />;
+
 export const SelectMultiple = (props: SelectMultipleProps) => {
   const { uploadedFile, fileList, setFileList } = props;
+  const [open, setOpen] = useState(false);
+  const closePopper = () => setOpen(false);
+  const openPopper = () => setOpen(true);
 
   const initialTags = getTagsForSlot(uploadedFile.id, fileList);
-  const [tags, setTags] = useState<string[]>(initialTags);
-  const previousTags = usePrevious(tags) || initialTags;
-  const [open, setOpen] = React.useState(false);
-
-  const handleChange = (event: SelectChangeEvent<typeof tags>) => {
-    const {
-      target: { value },
-    } = event;
-    setTags(
-      // On autofill we get a stringified value.
-      typeof value === "string" ? value.split(",") : value,
-    );
-  };
-  const handleClose = () => setOpen(false);
-  const handleOpen = () => setOpen(true);
 
-  const updateFileListWithTags = (
-    previousTags: string[] | undefined,
-    tags: string[],
+  /**
+   * Options for autocomplete
+   * FileList with appended "category" property for grouping
+   */
+  const options: Option[] = useMemo(
+    () =>
+      (Object.keys(fileList) as Array<keyof typeof fileList>)
+        .filter((fileListCategory) => fileList[fileListCategory].length > 0)
+        .flatMap((category) =>
+          fileList[category].map((fileType) => ({ category, ...fileType })),
+        ),
+    [fileList],
+  );
+
+  /**
+   * Previous values to pre-populate autocomplete
+   */
+  const value: Option[] = useMemo(
+    () =>
+      initialTags.flatMap((tag) => options.filter(({ name }) => name === tag)),
+    [initialTags],
+  );
+
+  /**
+   * Update FileList (from main Public.tsx component) when values are added or removed from Autocomplete
+   */
+  const handleChange = (
+    _event: React.SyntheticEvent,
+    value: Option[],
+    reason: AutocompleteChangeReason,
```

Comment:
Another nice DX bonus - the `Autocomplete` has a much nicer API for handling changes as it expects items to be added/removed/changed.